### PR TITLE
uploader mutations

### PR DESCRIPTION
**Pull request by jkao1**
This PR creates and modifies mutations to optimize the data flow between the API and the uploader.

### CODE REVIEW

File: app/graphql/resolvers/create_article.rb [hunk 1]
```diff
@@ -15,17 +19,26 @@ class Resolvers::CreateArticle < Resolvers::MutationFunction
   # args - are the arguments passed
   # _ctx - is the GraphQL context (which would be discussed later)
   def call(_obj, args, ctx)
-    validate_admin(ctx)
+    if !admin_is_valid(ctx)
```
Comment (NicholasLYang) [inline app/graphql/resolvers/create_article.rb]: weird phrasing

File: app/graphql/resolvers/create_article.rb [hunk 2]
```diff
@@ -15,17 +19,26 @@ class Resolvers::CreateArticle < Resolvers::MutationFunction
...
+      return GraphQL::ExecutionError.new("Invalid user token. Please log in.")
+    end
     @article = Article.new(
```
Comment (NicholasLYang) [inline app/graphql/resolvers/create_article.rb]: This whole thing needs to be in a transaction. That way if stuff fails, it won't leave the db in a malformed state.

File: app/graphql/resolvers/create_medium.rb [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+    if media_type == "illustration"
+      roleTitle = "Illustrator"
+    else      
+      roleTitle = "Photographer"
+    end
+    role = Role.find_by(title: roleTitle)
+
+    profile = Profile.find_by(role_id: role.id, user_id: args["user_id"])
+    if profile.nil?
```
Comment (NicholasLYang) [inline app/graphql/resolvers/create_medium.rb]: I believe there is a method find_or_create_by

File: app/graphql/resolvers/create_user.rb [hunk 4]
```diff
@@ -8,15 +8,28 @@ class Resolvers::CreateUser < GraphQL::Function
...
   type Types::UserType
 
-  def call(_obj, args, _ctx)
-    User.create!(
+  def call(_obj, args, ctx)    
+    if !admin_is_valid(ctx)
+      return GraphQL::ExecutionError.new("Invalid user token. Please log in.")
+    end
+
+    emailUser = User.find_by(email: args["email"])
```
Comment (NicholasLYang) [inline app/graphql/resolvers/create_user.rb]: little awkward for a name

Comment (NicholasLYang) [reply app/graphql/resolvers/create_user.rb]: also camelCase is not ruby convention

File: app/graphql/resolvers/create_article.rb (hunk 2 above)
Comment (jkao1) [reply app/graphql/resolvers/create_article.rb]: So you want me to surround everything after the first `if` statement in a `ActiveRecord::Base.transaction do ... end` ?

File: app/graphql/resolvers/create_user.rb (hunk 4 above)
Comment (jkao1) [reply app/graphql/resolvers/create_user.rb]: suggestions? I've thought hard about it, and it's what I settled for.

File: app/graphql/types/query_type.rb [hunk 5]
```diff
@@ -59,6 +59,16 @@
     resolve ->(obj, args, ctx) { User.find_by(slug: args["slug"])}
   end
 
+  field :userByFirstLast do
```
Comment (NicholasLYang) [inline app/graphql/types/query_type.rb]: FirstLast is not very clear. Either userByName or userByFirstLastName makes more sense imo

File: app/graphql/resolvers/create_article.rb (hunk 2 above)
Comment (NicholasLYang) [reply app/graphql/resolvers/create_article.rb]: yep pretty much

File: app/graphql/resolvers/create_medium.rb [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+  # return type from the mutation
+  type Types::MediumType
+
+  # the mutation method
+  # _obj - is parent object, which in this case is nil
+  # args - are the arguments passed
+  # _ctx - is the GraphQL context (which would be discussed later)
+  def call(_obj, args, ctx)
+    if !admin_is_valid(ctx)
```
Comment (NicholasLYang) [inline app/graphql/resolvers/create_medium.rb]: Not a huge deal but remember postfix if is a thing!

File: app/graphql/resolvers/create_user.rb (hunk 4 above)
Comment (NicholasLYang) [reply app/graphql/resolvers/create_user.rb]: preexisting_user? goddamn_user? I guess email_user is fine

**Pull request merged**